nixos/modules/services/web-apps/bookstack.nix

```diff
@@ -0,0 +1,352 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.bookstack;
+  bookstack = cfg.package;
+  db = cfg.database;
+  mail = cfg.mail;
+
+  user = cfg.user;
+  group = config.services.nginx.group;
```

Comment:
I would recommend a dedicated group for `bookstack`. By adding the `bookstack` user to the `nginx` group you're potentially giving `bookstack` more access than desirable.

If instead you flip this around and say `users.users.nginx.extraGroup = [ "bookstack" ];` this is better from a security point of view. We use this pattern in some NixOS modules already.

Reply:
I'm not adding bookstack to the nginx group. The `group` variable is used in a different way here. I'll change it's name to make that more clear.